sharedse/src/jvmTest/kotlin/com/ustadmobile/sharedse/util/UstadTestRuleSe.kt

```diff
@@ -74,7 +83,17 @@ class UstadTestRule: TestWatcher() {
 
             registerContextTranslator { account: UmAccount -> Endpoint(account.endpointUrl) }
 
-            bind<Gson>() with singleton { Gson() }
+            bind<Gson>() with singleton {
+                val builder = GsonBuilder()
+                builder.registerTypeAdapter(Statement::class.java, StatementSerializer())
+                builder.registerTypeAdapter(Statement::class.java, StatementDeserializer())
+                builder.registerTypeAdapter(ContextActivity::class.java, ContextDeserializer())
+                builder.create()
+            }
+
+            bind<XapiStatementEndpoint>() with singleton {
```

Comment:
Is this really needed for the test rule? Shouldn't any item that is going to use the XapiStatementEndpoint bind it in it's own test (e.g. using a mock).

If it is going to be used, it needs to use the context (e.g. as you have done in UstadApp.kt) used with scoped ... This would connect all xapi statements to one database.